source/resampling_with_code2.Rmd

```diff
@@ -0,0 +1,669 @@
+12 simulated jurors.
+
+## Using code for simulated trial
+
+::: {.python}
+
+We use the same logic to simulate a trial with the computer. A little code
+makes the job easier, because we can ask {{< var lang >}} to give us 12 random
+numbers from 0 through 99, and to count how many of these numbers are in the
+range from 75 through 99.  Numbers in the range from 75 through 99 correspond
+to Black jurors.
+
+### Ranges in {{< var lang >}}
+
+We said above, that ranges can be confusing, and we use "from 1 through 5" to avoid this confusion.
+
+::: {.python}
+Python and NumPy also have the concept of a range of integers (positive or
+negative whole numbers).  Here is a NumPy range:
+
+```{python, opts.label="py_ed"}
+# Load the NumPy library.
+import numpy as np
+
+# Make a range from 0 up to (not including) 6
+some_numbers = np.arange(0, 6)
+
+# Show the result
+some_numbers
+```
+
+Notice that `np.arange(0, 6)` means "from 0 through 5" — the range *does not
+include* the last value.  Another way of putting this is that `np.arange(0,
+6)` means "from 0 up to, but not including 6".
+
+This takes a little getting used to, but we think you will find it starts to
+come naturally after a while.
+:::
+
+::: {.r}
+R also has the concept of a range of integers (positive or negative whole
+numbers).  Here is an R range:
+
+```{r, opts.label="r_ed"}
+# Make a range from 0 through 5
+some_numbers <- 0:5
+
+# Show the result
+some_numbers
+```
+
+Notice that `0:5` means exactly "from 0 through 5" — the range includes all
+the integers starting at 0 and including 5.
+:::
+
+### Random numbers from 0 through 99
+
+We can now use {{< var np_or_r >}} and the random number function from the last chapter, to get 12 random numbers from 0 through 99.
+
+```{python, opts.label="py_ed"}
+# Ask NumPy for a random number generator.
+rnd = np.random.default_rng()
+
+# Get 12 random numbers from 0 through 99
+# (from 0, up to, but not including 100).
+zero_through_99 = np.arange(0, 100)
+a = rnd.choice(zero_through_99, 12)
+
+# Show the result
+a
+```
+
+```{r, opts.label="r_ed"}
+# Get 12 random numbers from 0 through 99
+a <- sample(0:99, 12, replace=TRUE)
+
+# Show the result
+a
+```
+
+### Counting the jurors
+
+We use *comparison* and [`np.sum`]{.python}[`sum`]{.r} to count how
+many numbers are are greater than 74, and therefore, in the range from 75
+through 99:
+
+```{python, opts.label="py_ed"}
+# How many numbers are greater than 74?
+b = np.sum(a > 74)
+# Show the result
+b
+```
+
+```{r, opts.label="r_ed"}
+# How many numbers are greater than 74?
+b <- sum(a == 9)
+# Show the result
+b
+```
+
+### A single simulated trial
+
+We assemble the pieces from the last few sections to make a {{< var cell >}} that simulated a single trial:
+
+```{python, opts.label="py_ed"}
+rnd = np.random.default_rng()
+# Get 12 random numbers from 0 through 99
+zero_through_99 = np.arange(0, 100)
+a = rnd.choice(zero_through_99, 12)
```

Comment:
`rng.integers(0, 100, size=12)`

Reply:
Well - yes - but then we have to introduce `rnd.integers`.  So I was thinking we could get on to `rnd.choice([0, 1], p=[0.74, 0.26])` soon, and therefore avoid the need for `integers`.  And then of course we have to explain the '100' there, which is just slightly annoying.